Sources/App/Onboarding/Container/OnboardingNavigationView.swift

```diff
@@ -0,0 +1,94 @@
+import Shared
+import SwiftUI
+
+enum OnboardingStyle: Equatable {
+    enum RequiredType: Equatable {
+        case full
+        case permissions
+    }
+
+    case initial
+    case required(RequiredType)
+    case secondary
+
+    var insertsCancelButton: Bool {
+        switch self {
+        case .initial, .required: return false
+        case .secondary: return true
+        }
+    }
+}
+
+enum OnboardingNavigation {
+    public static var requiredOnboardingStyle: OnboardingStyle? {
+        if Current.servers.all.isEmpty {
+            return .required(.full)
+        } else if !OnboardingPermissionHandler.notDeterminedPermissions.isEmpty {
+            return .required(.permissions)
+        } else {
+            return nil
+        }
+    }
+}
+
+struct OnboardingNavigationView: View {
+    static func controller(onboardingStyle: OnboardingStyle) -> UIViewController {
+        OnboardingNavigationView(onboardingStyle: onboardingStyle).embeddedInHostingController()
+    }
+
+    @Environment(\.dismiss) private var dismiss
+    @StateObject public var viewModel = OnboardingNavigationViewModel()
+    public let onboardingStyle: OnboardingStyle
+
+    init(onboardingStyle: OnboardingStyle) {
+        self.onboardingStyle = onboardingStyle
+    }
+
+    var body: some View {
+        NavigationView {
+            Group {
+                switch onboardingStyle {
+                case .initial:
+                    OnboardingWelcomeView(shouldDismissOnboarding: $viewModel.shouldDismiss)
+                case .secondary:
+                    OnboardingServersListView(shouldDismissOnboarding: $viewModel.shouldDismiss)
+                case let .required(type):
+                    switch type {
+                    case .full:
+                        OnboardingWelcomeView(shouldDismissOnboarding: $viewModel.shouldDismiss)
+                    case .permissions:
+                        OnboardingPermissionsNavigationView(onboardingServer: nil)
```

Comment:
```suggestion
                        OnboardingPermissionsView(onboardingServer: nil)
```
To fit the other views. Except if I didn't get the pattern here.

Reply:
`OnboardingPermissionsNavigationView` is not a permission view per say, it is a navigation container for all required permission views, thats why the naming.

Reply:
Sorry I missed you second comment and replied over it, yes for now location only, but this flow is not final, I removed motion and focus sensor, they may return before the release, this will stay flexible in case we have more permissions later.